app/src/ui/undo/warn-local-changes-before-undo.tsx

```diff
@@ -44,48 +50,66 @@ export class WarnLocalChangesBeforeUndo extends React.Component<
         onSubmit={this.onSubmit}
         onDismissed={this.props.onDismissed}
       >
-        <DialogContent>
-          <Row>{this.getWarningText()}</Row>
-        </DialogContent>
+        {this.getWarningDialog()}
         <DialogFooter>
           <OkCancelButtonGroup destructive={true} okButtonText="Continue" />
         </DialogFooter>
       </Dialog>
     )
   }
 
-  private getWarningText() {
-    if (
-      this.props.commit.isMergeCommit &&
-      !this.props.isWorkingDirectoryClean
-    ) {
+  private getWarningDialog() {
+    if (this.props.commit.isMergeCommit) {
+      return this.getMergeCommitWarningDialog()
+    } else {
       return (
-        <>
-          You have changes in progress. Undoing the merge commit might result in
-          some of these changes being lost.
-          <br />
-          <br />
-          {this.getMergeCommitUndoWarningText()}
-          <br />
-          <br />
-          Do you want to continue anyway?
-        </>
+        <DialogContent>
+          <Row>
+            You have changes in progress. Undoing the commit might result in
+            some of these changes being lost. Do you want to continue anyway?
+          </Row>
+          <Row>
+            <Checkbox
+              label="Do not show this message again"
+              value={
+                this.state.confirmUndoCommit
+                  ? CheckboxValue.Off
+                  : CheckboxValue.On
+              }
+              onChange={this.onConfirmUndoCommitChanged}
+            />
+          </Row>
+        </DialogContent>
       )
-    } else if (this.props.commit.isMergeCommit) {
+    }
+  }
```

Comment:
You can get rid of the `else` and just leave the `if` as a "fast return":
```typescript
    if (this.props.commit.isMergeCommit) {
      return this.getMergeCommitWarningDialog()
    }

    return ...
```

Reply:
refactored as you suggested